src/Data/Fin/Base.agda

```diff
@@ -310,6 +310,19 @@ compare (suc i) (suc j) with compare i j
 ... | greater greatest least = greater (suc greatest) (suc least)
 ... | equal   i              = equal   (suc i)
 
+data Ordering' (i : Fin n) : Fin n → Set where
```

Comment:
I'm pretty sure this is morally an instantiation of the existing `Trichotomous` type from `Relation.Binary.Definitions` (exported by `Relation.Binary`) and that `compare` is the existing `<-cmp` proof from `Data.Fin.Properties`?

Can we us those instead?

Reply:
I changed my code to use them instead.